native/src/main/java/io/ballerina/stdlib/sql/utils/RecordIteratorUtils.java

```diff
@@ -181,8 +180,7 @@ private static Object getResult(ResultSet resultSet, int columnIndex, ColumnDefi
                 Struct structData = (Struct) resultSet.getObject(columnIndex);
                 return resultParameterProcessor.convertStruct(structData, sqlType, ballerinaType);
             case Types.SQLXML:
-                SQLXML sqlxml = resultSet.getSQLXML(columnIndex);
-                return resultParameterProcessor.convertXml(sqlxml, sqlType, ballerinaType);
+                return resultParameterProcessor.getAndConvertXml(resultSet, columnIndex, sqlType, ballerinaType);
```

Comment:
Lets do the suggested changes for all data types in this switch statement.

Reply:
updated in https://github.com/ballerina-platform/module-ballerina-sql/pull/329/commits/ec7082e053a0528f35ea6b01e858df0b0ba0fb6f